docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,59 @@
+# Contributing
+
+When contributing to this repository, please first discuss the change you wish to make via issue, email, or any other method with the owners of this repository before making a change.
+
+## Roles
+
+Three roles are available for contributors and will lead to different tasks in the repository:
+
+- Integrator: adds a test suite, runs the suite, makes an issue for each failing test, marks failing tests as skipped.
+- Debuggor: picks an issue raised by a Integrator, and dives in the EVM spec and the current
+Cairo 0 implementation in order to pin point the exact issue. Ones the issue is correctly understood, checks if a similar issue hasn't been raised yet by another Debuggor yet. If it has, link the Integrator issue to the Debuggor issue for tracking and pick up another Integrator issue. If it hasn't, raises a detailed issue about the Cairo 0 implementation bug and links it to the Integrator issue.
+- Implementor: picks up an detailed issue raised by a Debuggor and reimplements the test logic in Cairo 0 python tests. Once the error is replicated, fixes the test in Cairo 0.
+
+## How to
+
+### Integrator
+
+In order to match the current EVM implementation from Reth, we want to run the same ethereum/tests as them. A list of all applicable tests can be found [here](https://github.com/paradigmxyz/reth/blob/main/testing/ef-tests/tests/tests.rs#L17).
+
+Check for an unassigned `epic` issue for a test. Add it to  `crates/ef-testing/tests.rs` by using the `blockchain_tests` macro. The first argument should be the name of your test, the second is the folder to find the test. Please use the snake case name of the folder for the name of the test (e.g. `blockchain_tests!(st_bad_opcode, stBadOpcode)` for adding stBadOpcode).
+
+Now comment all the other lines except for the test you added and run `make ef-tests`.
+After a while (count 10-15 minutes) the test will end and you should have in your terminal's output a list of all the failing tests. You can now start making issues. When raising issues, please try to match other `Integrator` issued on the style, and don't forget to add any raised issue to the `epic` issue's tasklist.
```

Comment:
Ah interesting, you think every single testsuite should be an epic, and every issue raised as a result of it should be present in its tasklist?
Today, I create both epic and features for adding testsuites, based on size. Open to challenges

Reply:
I find it more structured to have it this way, then you can easily go to the epic, which is linked to one testsuite, and find all issues related in it.

Reply:
Okay! will have to update my issues